analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/IOperationExtensions.cs

```diff
@@ -22,16 +22,18 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn;
 
 internal static class IOperationExtensions
 {
-    internal static ISymbol TrackedSymbol(this IOperation operation)
+    internal static ISymbol TrackedSymbol(this IOperation operation, ProgramState state)
     {
         return operation?.Kind switch
         {
-            OperationKindEx.Conversion when operation.ToConversion() is var conversion && !IsTryDownCast(conversion) => TrackedSymbol(conversion.Operand),
+            OperationKindEx.FlowCaptureReference => state.ResolveCapture(operation) is { Kind: not OperationKindEx.FlowCaptureReference } resolved
```

Comment:
Why stop on FlowCaptureRereference that references another flow capture? That's rather common. By removing the condition, it would recursively come back here and moved forward.

Reply:
This would create an endless loop if the capture could not get resolved. I guess we could check for the returned operation to be equal to the passed one.
Do you have an example of a `FlowCaptureReference` referencing another `FlowCaptureReference`?

Reply:
??= generates that scenario

Reply:
Yep, some other stop condition would be needed.

Reply:
The inner capture actually gets resolved when the outer capture gets stored:
https://github.com/SonarSource/sonar-dotnet/blob/fc6cc877d11c0ccd4267a24c3867b3ad5ed0c63c/analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/FlowCapture.cs#L29
But checking for equality looks cleaner, regardless.

Reply:
Using `when` should simplify it a bit, make it consistent with the rest, remove ternary and make it fit on a single line :)

Reply:
I didn't see an UT for the infinite recursion scenario.. or did I miss something? It will need a unrealistic ProgramState preparation

Reply:
I still don't see the UT for the infinite recursion scenario. Which one is it? I would expect an UT that will do `state.SetCapture` to itself. And then try to resolve it.

Reply:
The cause of the infinite recursion is that `state.ResolveCapture(operation)` returns `operation` if it cannot resolve the capture. This scenario is covered by `TrackedSymbol_UnknownCaptureReference_ReturnsNull`.